lib/base-compiler.js

```diff
@@ -55,6 +56,11 @@ Compile.prototype.newTempDir = function () {
 Compile.prototype.writeFile = Promise.denodeify(fs.writeFile);
 Compile.prototype.readFile = Promise.denodeify(fs.readFile);
 Compile.prototype.stat = Promise.denodeify(fs.stat);
+Compile.prototype.couldHaveOptOutput = function(options, version) {
+    //TODO(jared): a compiler meta data gatherer could improve this
+    return version.toLowerCase().indexOf("clang") > -1 && 
+          options.some((x) => x == "-fsave-optimization-record");
```

Comment:
If it helps sway your decision... I went looking for a gcc/clang object on the first pass through to try and figure out the best way to support this.